File: client/src/app/modules/tiers/services/tiers.followup.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { TiersFollowupType } from '../../tiers/model/TiersFollowupType';

@Injectable({
  providedIn: 'root'
})
export class TiersFollowupTypeService extends AppRestService<TiersFollowupType>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getTiersFollowupTypes() {
    return this.getListCached(new HttpParams(), "tiers-followup-types");
  }

  addOrUpdateTiersFollowupType(tiersFollowupType: TiersFollowupType) {
    this.clearListCache(new HttpParams(), "tiers-followup-types");
    return this.addOrUpdate(new HttpParams(), "tiers-followup-type", tiersFollowupType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/tiers/service/TiersFollowupTypeService.java
package com.jss.osiris.modules.osiris.tiers.service;

import java.util.List;

import com.jss.osiris.modules.osiris.tiers.model.TiersFollowupType;

public interface TiersFollowupTypeService {
    public List<TiersFollowupType> getTiersFollowupTypes();

    public TiersFollowupType getTiersFollowupType(Integer id);

    public TiersFollowupType addOrUpdateTiersFollowupType(TiersFollowupType tiersFollowupType);
}
